.github/workflows/build-images-branches.yaml

```
@@ -7,13 +7,72 @@ on:
+          echo "channels=${CHANNELS}" >> $GITHUB_OUTPUT
+          echo "defaultChannel=${DEFAULT_CHANNEL}" >> $GITHUB_OUTPUT
+
   build-branches:
     name: Calls build-images-base workflow
+    needs: detect-config
     uses: ./.github/workflows/build-images-base.yaml
-    secrets: inherit
+    # Only pass secrets actually required by build-images-base.yaml (principle of least privilege)
+    # These are used for docker login to push images to the container registry
+    secrets:
+      IMG_REGISTRY_USERNAME: ${{ secrets.IMG_REGISTRY_USERNAME }}
```

actionlint (1.7.12)

[error] secret "IMG_REGISTRY_USERNAME" is not defined in "./.github/workflows/build-images-base.yaml" reusable workflow. no secret is defined

(workflow-call)